flink-ml-lib/src/main/java/org/apache/flink/ml/feature/minmaxscaler/MinMaxScalerModel.java

```diff
@@ -0,0 +1,181 @@
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+        ReadWriteUtils.saveModelData(
+                MinMaxScalerModelData.getModelDataStream(modelDataTable),
+                path,
+                new MinMaxScalerModelData.ModelDataEncoder());
+    }
+
+    /**
+     * Loads model data from path.
+     *
+     * @param env Stream execution environment.
+     * @param path Model path.
+     * @return MinMaxScalerModel model.
+     */
+    public static MinMaxScalerModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        MinMaxScalerModel model = ReadWriteUtils.loadStageParam(path);
+        DataStream<MinMaxScalerModelData> modelData =
+                ReadWriteUtils.loadModelData(
+                        env, path, new MinMaxScalerModelData.ModelDataDecoder());
+        return model.setModelData(tEnv.fromDataStream(modelData));
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+        private final String featureCol;
+        private MinMaxScalerModelData minMaxScalerModelData;
+        private final double upperBound;
+        private final double lowerBound;
+        private final String broadcastKey;
+        private DenseVector maxVector;
+        private DenseVector minVector;
+
+        public PredictOutputFunction(
+                String broadcastKey, double upperBound, double lowerBound, String featureCol) {
+            this.upperBound = upperBound;
+            this.lowerBound = lowerBound;
+            this.broadcastKey = broadcastKey;
+            this.featureCol = featureCol;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (minMaxScalerModelData == null) {
+                minMaxScalerModelData =
+                        (MinMaxScalerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+                maxVector = minMaxScalerModelData.maxVector;
+                minVector = minMaxScalerModelData.minVector;
+            }
+            DenseVector feature = (DenseVector) row.getField(featureCol);
+            DenseVector outputVector = new DenseVector(maxVector.size());
+            if (feature != null) {
+                for (int i = 0; i < maxVector.size(); ++i) {
+                    if ((minVector.values[i] - maxVector.values[i]) != 0.0) {
+                        outputVector.values[i] =
+                                (feature.values[i] - minVector.values[i])
+                                                / (maxVector.values[i] - minVector.values[i])
```

Comment:
I think spanVector maybe not very good. 
1. In Spark and alink, maxVector and minVector are saved for modelData. 
2. If we spanVector, code may be not clear as minVector and maxVector used.

Reply:
I'm not sure this is the case for Spark. In Spark's MinMaxScaler.scala I found the following code:
```scala
values(i) = (values(i) - minArray(i)) * scaleArray(i) + minValue
```
which shows that Spark uses `minArray` and `scaleArray`, instead of `minArray` and `maxArray`.

Reply:
val scaleArray = Array.tabulate(numFeatures) { i =>
      val range = originalMax(i) - originalMin(i)

Reply:
Sounds good.